games/minesweeper.kak

```diff
@@ -0,0 +1,124 @@
+# kakoune implementation of the minesweeper game
+
+def -params 3 -docstring %{minesweeper <lines> <columns> <mines>: play a game of minesweeper in a grid of size <lines>x<columns> containing <mines> mines
+press <space> to uncover a square
+press f to flag it} \
+minesweeper %{
+    try %{ minesweeper_quit }
+    edit -scratch *minesweeper_solution*
+    # create $1*$2 lines, $3 with 'x' and the rest with ' ', shuffled
+    exec %sh{ printf %d $(($1 * $2)) }O<space><esc>gjdgg %arg{3} JK<a-s>Hrx\%|shuf<ret>
+    # join them in a $1x$2 grid
+    exec \%s(.\n){ %arg{2} }<ret>K<a-s>ld
+    # surround them by #
+    exec '%<a-s>i#<esc>a#<esc><space>xypxHr#LyggP'
+    # copy that layout to the play buffer
+    exec '%"my'
+    edit -scratch *minesweeper*
+    # but remove all 'x' here
+    exec -draft '"mpd%sx<ret>r<space>'
+
+    add-highlighter group minesweeper
+    add-highlighter -group minesweeper regex '\d' 0:black,green
+    add-highlighter -group minesweeper regex 'x' 0:black,red
+    add-highlighter -group minesweeper regex '!' 0:black,yellow
```

Comment:
I have `try %{ minesweeper_quit }` at the beginning of the command for that. In any case, I've checked that you can restart a game if one is already open.